sdk/keyvault/Azure.Security.KeyVault.Keys/samples/Sample5_SignVerify.md

```diff
@@ -0,0 +1,117 @@
+
+In the sample below, you can set `keyVaultUrl` based on an environment variable, configuration setting, or any way that works for your application.
+
+```C# Snippet:KeysSample5KeyClient
+var keyClient = new KeyClient(new Uri(keyVaultUrl), new DefaultAzureCredential());
+```
+
+## Creating keys
+
+First we'll create both an RSA key and an EC key which will be used to sign and verify.
+
+```C# Snippet:KeysSample5CreateKey
+string rsaKeyName = $"CloudRsaKey-{Guid.NewGuid()}";
+var rsaKey = new CreateRsaKeyOptions(rsaKeyName, hardwareProtected: false)
+{
+    KeySize = 2048,
+};
+
+string ecKeyName = $"CloudEcKey-{Guid.NewGuid()}";
+var ecKey = new CreateEcKeyOptions(ecKeyName, hardwareProtected: false)
+{
+    CurveName = KeyCurveName.P256K,
+};
+
+KeyVaultKey cloudRsaKey = keyClient.CreateRsaKey(rsaKey);
+Debug.WriteLine($"Key is returned with name {cloudRsaKey.Name} and type {cloudRsaKey.KeyType}");
+
+KeyVaultKey cloudEcKey = keyClient.CreateEcKey(ecKey);
+Debug.WriteLine($"Key is returned with name {cloudEcKey.Name} and type {cloudEcKey.KeyType}");
+```
+
+## Creating CryptographyClients
+
+Then we create the `CryptographyClient` which can perform cryptographic operations with the key we just created. Again we are using the default Azure credential as above.
+
+```C# Snippet:KeysSample5CryptographyClient
+var rsaCryptoClient = new CryptographyClient(cloudRsaKey.Id, new DefaultAzureCredential());
+
+var ecCryptoClient = new CryptographyClient(cloudEcKey.Id, new DefaultAzureCredential());
+```
+
+## Signing keys with the Sign and Verify methods
+
+Next we'll sign some arbitrary data and verify the signatures using the `CryptographyClient` with both the EC and RSA keys we created.
+The `Sign` and `Verify` methods expect a precalculated digest, and the digest needs to be calculated using the hash algorithm which matches the signature algorithm being used.
+SHA256 is the hash algorithm used for both RS256 and ES256K which are the algorithms we'll be using in this sample.
+
+```C# Snippet:KeysSample5SignKey
+byte[] data = Encoding.UTF8.GetBytes("This is some sample data which we will use to demonstrate sign and verify");
+byte[] digest = null;
+
+using (HashAlgorithm hashAlgo = SHA256.Create())
+{
+    digest = hashAlgo.ComputeHash(data);
+}
+
+SignResult rsaSignResult = rsaCryptoClient.Sign(SignatureAlgorithm.RS256, digest);
+Debug.WriteLine($"Signed digest using the algorithm {rsaSignResult.Algorithm}, with key {rsaSignResult.KeyId}. The resulting signature is {Convert.ToBase64String(rsaSignResult.Signature)}");
+
+SignResult ecSignResult = ecCryptoClient.Sign(SignatureAlgorithm.ES256K, digest);
+Debug.WriteLine($"Signed digest using the algorithm {ecSignResult.Algorithm}, with key {ecSignResult.KeyId}. The resulting signature is {Convert.ToBase64String(ecSignResult.Signature)}");
+```
+
+## Verifying signatures
+
+Verify the signatures.
```

Comment:
This statement is a repeat of the header.  Other statements expand on the header.  I prefer to be consistent.  Consider making more verbose.